### PR TITLE
Better validation in JSONL datasets

### PR DESCRIPTION
**Pull request by KUNAL1612**
**Patch Description**
Added code to check for validity of json in jsonl files in the index building function. Also added two new tests to make sure that empty jsonl files as well as corrupt jsonl files throw up the right errors.

**Testing steps**
ran the new tests to ensure that it works as expected

### CODE REVIEW

File: metaseq/data/jsonl_dataset.py [hunk 1]
```diff
@@ -88,6 +88,13 @@ def _build_index(self, path: str):
         cur = 0
         while True:
             line = f.readline()
+            if line != b"":
```
Comment (stephenroller) [inline metaseq/data/jsonl_dataset.py:91]: why is this bytes?

File: metaseq/data/jsonl_dataset.py [hunk 2]
```diff
@@ -88,6 +88,13 @@ def _build_index(self, path: str):
...
+                try:
+                    json.loads(line)
+                except json.decoder.JSONDecodeError:
+                    raise AssertionError(
+                        "Unable to load truncated or corrupted JSON file"
```
Comment (stephenroller) [inline metaseq/data/jsonl_dataset.py:96]: Instead of an assertion error, let's keep with JSONDecodeError but provide a better error message. Include the name of the file and the line number in the error message, please

**Comment by stephenroller** [comment]
Great tests!

File: metaseq/data/jsonl_dataset.py (hunk 1 above)
Comment (KUNAL1612) [reply metaseq/data/jsonl_dataset.py:91]: I noticed that that's just how line was formatted. And if line was b"" then it would always fail json loading anyway

File: metaseq/data/jsonl_dataset.py (hunk 2 above)
Comment (KUNAL1612) [reply metaseq/data/jsonl_dataset.py]: Noted. Updated the file to do this.

**Review by punitkoura — APPROVED** [review]
